src/PaymentProcessing/SpherePayment.php

```diff
@@ -0,0 +1,257 @@
+    }
+
+    /**
+     * Return sphere code in javascript block.
+     *
+     * @return string
+     */
+    public function renderSphereJs(): string
+    {
+        if ($this->front == 'patient') {
+            return '<script>' . $this->renderSphereJsPatientFront() . $this->renderSphereJsCore() . '</script>';
+        } else { // $this->front == 'clinic'
+            return '<script>' . $this->renderSphereJsClinicFront() . $this->renderSphereJsCore() . '</script>';
+        }
+    }
+
+    /**
+     * Return patient front specific sphere code in javascript block.
+     *
+     * @return string
+     */
+    private function renderSphereJsPatientFront(): string
+    {
+        return "
+            function sphereSuccess(encData) {
+                let oForm = document.forms['payment-form'];
+                oForm.elements['mode'].value = 'Sphere';
+
+                let inv_values = JSON.stringify(getFormObj('invoiceForm'));
+                document.getElementById('invValues').value = inv_values;
+
+                let hiddenInput = document.createElement('input');
+                hiddenInput.setAttribute('type', 'hidden');
+                hiddenInput.setAttribute('name', 'enc_data');
+                hiddenInput.setAttribute('value', encData);
+                oForm.appendChild(hiddenInput);
+
+                // Submit payment to server
+                fetch('./lib/paylib.php', {
+                    method: 'POST',
+                    body: new FormData(oForm)
+                }).then(function(response) {
+                    if (!response.ok) {
+                        throw Error(response.statusText);
+                    }
+                    return response.text();
+                }).then(function(data) {
+                    if(data !== 'ok') {
+                        alert(data);
+                        return;
+                    }
+                    alert(chargeMsg);
+                    window.location.reload(false);
+                }).catch(function(error) {
+                    alert(error)
+                });
+            }
+            ";
+    }
+
+    /**
+     * Return clinic front specific sphere code in javascript block.
+     *
+     * @return string
+     */
+    private function renderSphereJsClinicFront(): string
+    {
+        return "
+            function sphereSuccess(authCode, transId) {
+                $('#check_number').val(transId);
+                alert(" . xlj('Success') . " + '\\n' + 'Auth: ' + authCode + ' TransId: ' + transId);
+                // core payment process
+                $(\"[name='form_save']\").click();
+            }
+            ";
+    }
+
+    /**
+     * Return core sphere code in javascript block.
+     *   Will employ the front and production status in the rendering.
+     *
+     * @return string
+     */
+    private function renderSphereJsCore(): string
+    {
+        return "
+            function sphereNotSuccess(message) {
+                alert(message);
```

Comment:
Are we logging these messages anywhere?  Or does Sphere backend log these messages for the provider/IT go and look them up?  

I've always found that good payment processing contains a log of the errors so you can quickly fix and resolve issues.